Simplenote/src/main/java/com/automattic/simplenote/utils/ReviewAccountVerifier.java

```diff
@@ -0,0 +1,74 @@
+package com.automattic.simplenote.utils;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.automattic.simplenote.Simplenote;
+import com.automattic.simplenote.models.Account;
+import com.simperium.client.Bucket;
+import com.simperium.client.BucketObjectMissingException;
+
+import static com.automattic.simplenote.models.Account.KEY_EMAIL_VERIFICATION;
+
+/**
+ * When a user login into their account, Simplenote starts indexing the different buckets including Bucket<Account>.
+ * We setup a OnNetworkChangeListener on the account bucket to listen when the bucket is indexed. One of the parameters
+ * of the callback onNetworkChange called is Bucket.ChangeType. If this parameter is INDEX, then we know that the
+ * account bucket was indexed (synchronized). When the account bucket is synchronized, we can check whether the account
+ * was verified or not
+ */
+public class ReviewAccountVerifier implements Bucket.OnNetworkChangeListener<Account> {
+    private enum Status {
+        UNKNOWN,
+        UNVERIFIED,
+        VERIFIED,
+    }
+
+    private final Simplenote simplenote;
+    private Status currentStatus = Status.UNKNOWN;
+
+    public ReviewAccountVerifier(Simplenote simplenote) {
+        this.simplenote = simplenote;
+    }
+
+    public void clearStatus() {
+        currentStatus = Status.UNKNOWN;
+    }
+
+    @Override
+    public void onNetworkChange(Bucket<Account> bucket, Bucket.ChangeType type, String key) {
```

Comment:
this comes from my own personal style but I like to avoid implementing these interfaces directly on classes that contain their own behavior. that is, by removing the boundary between the account review behavior and the network listener behavior. this is a small class so it won't make too much of an impact here, but for example, in [SyncTimes](https://github.com/Automattic/simplenote-android/blob/develop/Simplenote/src/main/java/com/automattic/simplenote/SyncTimes.java#L60) I created a listener as a property of the class and [connected its listener](https://github.com/Automattic/simplenote-android/blob/develop/Simplenote/src/main/java/com/automattic/simplenote/Simplenote.java#L126) vs. connecting itself as `this`. I found this quite helpful in separating the code into pieces that felt independent from each other.

```java
mVerificationWatcher = new AccountVerificationWatcher(this);
mAccountBucket.addOnNetworkChangeListener(mVerificationWatcher.bucketListener);
```

this kind of separation duly pays off when we want to have a class listen to more than one bucket but has its own benefits otherwise by splitting the "new event from the bucket" logic from the "state of the verification status" logic.